packages/ui/src/services/utils.ts

```diff
@@ -75,10 +75,13 @@ export const getValueByPath = <T extends Record<string, unknown>>(option: T, pro
  * Finds value of nested property inside of an object.
  *
  * @param option - Object to look properties inside
- * @param prop - string or function used to find nested property
+ * @param prop - string, number or function used to find nested property
  */
-export const getProp = <T extends (Record<string, unknown> | string)> (option: T, prop: string | ((t: T) => any) | Function): any => {
-  if (typeof option === 'string') { return }
+export const getValueByKey = <T extends (Record<string, unknown> | string | number)> (
+  option: T,
+  prop: string | ((t: T) => any),
+): any => {
```

Comment:
```suggestion
export const getValueByKey = <R extends unknown, T extends (Record<string, R> | string | number)> (
  option: T,
  prop: string | ((t: T) => R),
): R=> {
```

Let's have return type annotated.